reVX/least_cost_xmission/layers/friction_barrier_builder.py

```diff
@@ -7,58 +7,61 @@
 
 import numpy as np
 import numpy.typing as npt
-from reVX.config.transmission_layer_creation import Extents, FBLayerConfig
 
+from reVX.handlers.layered_h5 import LayeredTransmissionH5
+from reVX.config.transmission_layer_creation import Extents, FBLayerConfig
+from reVX.least_cost_xmission.layers.base import LayerCreator
 from reVX.least_cost_xmission.layers.utils import rasterize_shape_file
 from reVX.least_cost_xmission.layers.masks import MaskArr, Masks
 from reVX.least_cost_xmission.config.constants import (DEFAULT_DTYPE,
                                                        RAW_BARRIER_TIFF,
                                                        FRICTION_TIFF)
-from reVX.least_cost_xmission.layers.transmission_layer_io_handler import (
-    TransLayerIoHandler
-)
 
 logger = logging.getLogger(__name__)
 
 ALL = 'all'
 
 
-class FrictionBarrierBuilder:
+class FrictionBarrierBuilder(LayerCreator):
     """
     Build friction or barrier layers.
     """
-    def __init__(self, type_: Literal['friction', 'barrier'],
-                 io_handler: TransLayerIoHandler, masks: Masks,
+    def __init__(self, io_handler: LayeredTransmissionH5,
+                 mask: Masks, output_tiff_dir=".",
                  dtype: npt.DTypeLike = DEFAULT_DTYPE):
         """
         Parameters
         ----------
-        type_ : {'friction', 'barrier'}
-            Type of layer being built
-        io_handler : TransLayerIoHandler
-            IO handler
-        masks : Masks
-            Mask Handler
+        io_handler : :class:`LayeredTransmissionH5`
+            Transmission layer IO handler
+        mask : Masks
+            Masks instance that can be used to retrieve multiple types
+            of masks.
+        output_tiff_dir : path-like, optional
+            Directory where cost layers should be saved as GeoTIFF.
+            By default, ``"."``.
         dtype : np.dtype, optional
             Data type for final dataset. By default, ``float32``.
         """
-        self._type = type_
-        self._io_handler = io_handler
-        self._masks = masks
-        self._dtype = dtype
+        super().__init__(io_handler=io_handler, mask=mask,
```

Comment:
docstring for `mask` states `ndarray` in `LayerCreator` may be out of data. Passing a `Masks` instance here

Reply:
Thinking about this more, I'm tempted to say make `mask` optional in `LayerCreator`, and create a `self._masks: Masks` attribute just for `FrictionBarrierBuilder` so there is no confusion. Or get a little fancier and allow both on `LayerCreator` but FBB seems to be the only subclass that needs it.

Reply:
Yea, this was the trickiest part - I tried to get rid of the `Masks` dependency for FBB but couldn't do it. BUT, this is the only difference between FBB and the `LayerCreator` base, so I still went with it.

I do think your idea of making it optional may be the best bet, so I will do that!

Reply:
```
class FrictionBarrierBuilder(LayerCreator):
    def __init__(self,  masks: Masks, ...):
        super().__init__(io_handler=io_handler,
                         output_tiff_dir=output_tiff_dir, dtype=dtype)
        self._masks: Masks = masks
```

Reply:
Something like that? possible confusion between `_mask` and `_masks`, but I'm ok with it.

Reply:
yup, that's pretty much exactly what I threw in there!